quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/jaxrs/TransactionalResponseInterceptor.java

```diff
@@ -34,13 +35,21 @@
 @ConstrainedTo(RuntimeType.SERVER)
 @Priority(10000)
 public class TransactionalResponseInterceptor implements WriterInterceptor, TransactionalSessionHandler {
+
     @Override
-    public void aroundWriteTo(WriterInterceptorContext context) throws IOException, WebApplicationException {
+    public void aroundWriteTo(WriterInterceptorContext context) {
         KeycloakSession session = Resteasy.getContextData(KeycloakSession.class);
+        OutputStream outputStream = context.getOutputStream();
 
-        try {
+        // see https://github.com/keycloak/keycloak/issues/20451
+        try (ByteArrayOutputStream buffer = new ByteArrayOutputStream(8192)) {
```

Comment:
I am curious how you chosen this particular number 8192? Is it what quarkus is using?

My 1st impression might be to use smaller number as default ByteArrayOutputStream uses 32 by default and using smaller size might be fine as the size should automatically increase if needed. However I also see that too small size might cause some additional overhead for copying the buffers and hence might not be ideal as well...

Just wondering...

Reply:
`BufferedInputStream` uses `8192` as the default. I think it is a reasonable default (perhaps even lower) considering how much data we expect in Keycloak responses.

We are still discussing the proper solution though. Please, see the issue.

Reply:
Thanks, I am resolving this.